### DISCUSSION TITLE
NeuroDesk general data security query

### DISCUSSION THREAD

**Discussion by ghost** [neurodesk/neurodesk.github.io-hugo#260] [Q&A] [answered]
Dear experts,

I have been using NeuroDesk on a browser to visualise my data and run some simple commands - it’s a wonderful and handy project I must say! I’ve enjoyed the ease of not having to go through 1000s of loopholes to try and install software on my own systems (as they all come with their set of dependencies).

My main concern is - where is the data stored? I understand that if I use NeuroDesk Play on a Browser and subsequently close the browser windows, the data gets deleted. But where is it stored in the meantime, and is this breaching any data security protocols - when I upload the data there is no identifiable string in the file names they’re just called T1.mgz aseg.mgz etc. and the data is coded, but I’m not sure if anything can be extracted from the metadata?

Just trying to be super careful here, as I do want to continue using this and unfortunately, the system admin won’t let me download any software locally on the university remote desktop from where I am opening up the data and subsequently transferring to the NeuroDesk browser so I don’t have sudo privileges. Accessing NeuroDesk on a browser seems like the best option, as I'm running all the hard processing on an HPC cluster system which has no graphical interface. 

In essence, I just want to look at my Freesurfer outputs on the university system somehow to prevent a breach of data security in any shape or form - but it's become oddly complicated to view them without Freeview... 

Any help is appreciated!

**Comment by stebo85** [comment, marked as answer]
Dear @aalam-23,

Great to hear Neurodesk solves problems for you :) 

To first answer your question: The data is stored in the Oracle Cloud - depending on the region you are using the Play service:
US-EAST: Ashburn
US-West: Phoenix
Germany-Central: Frankfurt
Australia-East: Sydney

Correct, the data is deleted after you close your browser window when using the Play/Binder service within at least 4 hours, but if your ethics / university doesn't allow you to process your anonymised data in the cloud, then this would still not be ideal.

As an alternative: You can run Neurodesk in your own institution quite easily. Depending on your setup, here a few ideas:
1) You can download the freesurfer singularity container to your HPC (https://www.neurodesk.org/docs/neurocontainers/singularity/). For this your HPC only needs singularity installed, which most HPCs have these days. Then connect to the HPC via X-Forwarding (ssh -X user@hpc). Now you only need an X-server on your machine (built into linux or mobaxterm for Windows) and you can open freeview directly on your HPC - the GUI get's displayed on the machine where you are connecting from.

2) You could install docker on your Remote desktop and run the full Neurodesk desktop interface there: https://www.neurodesk.org/docs/neurodesktop/getting-started/linux/

3) You could install singularity on your Remote desktop and run the individual neurocontainers from there. If you have an X-server installed this will work quite nicely.

4) We are working on another way of accessing graphical applications from an HPC in a browser, but this will still take a few weeks to finish - let me know if nothing from the above works for you and you can be an early test case for this new feature.

Kind regards
Steffen

**Reply by ghost** [reply]
Hi @stebo85,

Thank you very much for your detailed explanation!
I finally found a way to X-forward efficiently on my system and ended up looking at the scans using the Freesurfer version that was already installed on the HPC cluster and it has worked quite well for me!

However, I will keep NeuroDesk in mind as a backup option in case anything goes wrong in the future. Thank you so much for the work you're doing, it's saving a lot of us from spending hours trying to install these packages from scratch.
